packages/react-native/scripts/ios-prebuild/__docs__/README.md

```diff
@@ -0,0 +1,137 @@
+| `--compose`   | `-c`  | boolean | -                                          | Compose XCFramework from built dependencies                         |
+| `--platforms` | `-p`  | array   | `['ios', 'ios-simulator', 'mac-catalyst']` | Specify one or more platforms to build for                          |
+| `--flavor`    | `-f`  | string  | `Debug`                                    | Specify the flavor to build: `Debug` or `Release`                   |
+| `--identity`  | `-i`  | string  | -                                          | Specify the code signing identity to use for signing the frameworks |
+| `--help`      | -     | boolean | -                                          | Show help information                                               |
+
+### Output Structure
+
+The build produces:
+
+- XCFrameworks in the specified output directory
+- Debug symbols (dSYM files) for debugging
+- Build products organized by platform and configuration
+
+## Architecture
+
+The build system consists of several components:
+
+### `cli.js`
+
+The main entry point that orchestrates the build process. It:
+
+- Parses command-line arguments
+- Validates build parameters
+- Coordinates the build, archiving, and XCFramework creation steps
+
+### `build.js`
+
+Handles the Swift Package build process. It:
+
+- Executes `xcodebuild` commands with appropriate flags
+- Builds for specific platforms and build types (Debug/Release)
+- Locates and validates the generated framework artifacts
+- Uses build settings like `BUILD_LIBRARY_FOR_DISTRIBUTION=YES` for binary
+  compatibility
+
+### `types.js`
+
+Defines TypeScript/Flow type definitions for:
+
+- `BuildFlavor`: Debug or Release configurations
+- `Destination`: Target platforms (iOS, iOS Simulator, Catalyst, Vision,
+  visionOS)
+- `ArchiveOptions`: Configuration options for the build process
+
+### `utils.js`
+
+Provides utility functions including:
+
+- Logging functionality with prefixed output
+- Common helper functions used across scripts
+
+## Build Flags
+
+The build process uses specific `xcodebuild` flags:
+
+- `BUILD_LIBRARY_FOR_DISTRIBUTION=YES`: Enables module stability
+- `SKIP_INSTALL=NO`: Ensures frameworks are properly installed
+- `DEBUG_INFORMATION_FORMAT="dwarf-with-dsym"`: Generates debug symbols
+- `OTHER_SWIFT_FLAGS="-no-verify-emitted-module-interface"`: Skips interface
+  verification (useful for React Native modules due to the header structure not
+  beeing modular)
+
+## Notes
+
+- These scripts build React Native itself, not third-party dependencies
+- The build process requires significant disk space for derived data
+- Build times vary depending on the target platform and configuration
+- XCFrameworks support multiple architectures in a single bundle
+
+## Known Issues
+
+The generated XCFrameworks currently use CocoaPods-style header structures
+rather than standard framework header conventions. This may cause modularity
+issues when:
```

Comment:
```suggestion
The generated XCFrameworks currently use CocoaPods-style header structures rather than standard framework header conventions. This may cause modularity issues when:
```